replica/compaction_group.hh

```diff
@@ -222,12 +222,51 @@ public:
 using storage_group_vector = utils::chunked_vector<std::unique_ptr<storage_group>>;
```

Comment:
Please note that this commit introduces construction loop:

In `table` constructor we create 
```
, _sg_manager(make_storage_group_manager())
```

In `make_storage_group_manager` we create `tablet_storage_group_manager/single_storage_group_manager`.
In their constructors we create `compaction_group` which in turn calls `compaction_manager::add` in its constructor:
```
void compaction_manager::add(table_state& t) {
    auto [_, inserted] = _compaction_state.try_emplace(&t, t);
    if (!inserted) {
        on_internal_error(cmlog, format("compaction_state for table {} [{}] already exists", t, fmt::ptr(&t)));
    }
}
```
If table_state isn't inserted properly we hit:
https://github.com/scylladb/scylladb/blob/e3de30f94392672fa858df39e42735eb7b58662e/compaction/table_state.hh#L62-L73

which uses
```
table_state::get_group_id()  ->
table::compaction_groups() ->
storage_group_manager::compaction_groups()
```

And `storage_group_manager` is being created at this point so it's nullptr.

Reply:
thanks.  will fix in next version.